server/handlers.go

```diff
@@ -585,8 +595,10 @@ func (ah *ActivityHandler) handleDeletedActivity(activityIds clientmodels.Activi
 		return metrics.DiscardedReasonOther
 	}
 
+	ah.IgnorePluginHooksMap.Store(fmt.Sprintf("delete_post_%s", postInfo.MattermostID), true)
```

Comment:
Switching to the new unit tests exposed a race condition that the plugin elsewhere solves using the `IgnorePluginHooksMap` -- the idea is that when we call `DeletePost` below, the server will reflect that event back to us as a message hook, and we'll try to delete the Teams post (which will in turn fail because it's already deleted). I'm uncertain still if this is the right long term solution, but adding it here to mimic what we already do for creating posts and reactions.